### DISCUSSION TITLE
Using Pinia within Server/Api

### DISCUSSION THREAD

**Discussion by lazercaveman** [nuxt/nuxt#16627] [Questions] [answered]
Hello I was working with nuxt 3 and Pinia - using pinia within the server/api folder, which works fine in dev but will lead to an error when starting the build - is Pinia even thought to be used that way? (the documentation mentions SSR with nuxt, but seems to be a bit vague on that to me) … so the question is, is it legit using Pinia that way or should it only be used within pages?

**Comment by sevillaarvin** [comment]
What's the error? Can you provide a reproduction repo?

**Reply by lazercaveman** [reply]
Yes of course. : )

Builds generated from Nuxt currently crashes (nuxt instance unavailable), whereas Nuxt can be run via dev without problems - anyways, the problem seems to be related to Pinia:

```bash
[nuxt] [request error] [unhandled] [500] Cannot find module '/Volumes/nuxt/.output/server/node_modules/pinia/dist/pinia' imported from /Volumes/nuxt/.output/server/chunks/page.mjs
  at new NodeError (internal/errors.js:322:7)  
  at finalizeResolution (internal/modules/esm/resolve.js:318:11)  
  at moduleResolve (internal/modules/esm/resolve.js:776:10)  
  at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:887:11)  
  at Loader.resolve (internal/modules/esm/loader.js:89:40)  
  at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)  
  at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:76:40)  
  at link (internal/modules/esm/module_job.js:75:36)
```

After removing pinia entirely Nuxt works just fine - I don't know where the problem exactly is located - setup seems fine to me:

I made an example on stackblitz as well : ) 
https://stackblitz.com/edit/nuxt-starter-2cyqru?file=README.md,server%2Fapi%2Fpage.ts

**Reply by sevillaarvin** [reply]
Try filing it as an issue.

I can't figure out what's wrong, here are some references:

https://github.com/vuejs/pinia/issues/1688
https://github.com/nuxt/framework/discussions/6806

**Reply by sevillaarvin** [reply]
I created an issue here with a minimal reproduction: https://github.com/nuxt/framework/issues/7910

**Comment by SleepiestAdam** [comment]
Running into the same issue in my own project!

**Comment by lazercaveman** [comment, marked as answer]
It isn't working because the feature isn't implemented:
https://github.com/nuxt/framework/issues/7910#issuecomment-1262593378
